File: level1/392/is_subsequence.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

bool isSubsequence(char *s, char *t){
	int len = strlen(s);
	int	i = 0;
	int	j = 0;
	int	 counter = 0;

	while (t[j] && s[i])
	{
		if (t[j] == s[i])
		{
			counter++;
			i++;
		}
		j++;
	}
	if (counter == len)
		return (true);
	return (false);
}


// int main(void)
// {
// 	char *s = "axc";
// 	char *t = "ahbgdc";
// 	int x = isSubsequence(s, t);
// 	printf("%s", x ? "true" : "false");
// 	return (0);
// }

File: level1/392/is_subsequence.py
class Solution(object):
	def isSubsequence(self, s, t):
		len_s, len_t = len(s),len(t)
		i = 0
		j = 0
		while i < len_s and j < len_t:
			if s[i] == t[j]:
				i += 1
			j += 1
		return len_s == i

def main():
	s = "axc"
	t = "ahbgdc"
	solution = Solution()
	print(solution.isSubsequence(s, t))

if __name__ == "__main__":
	main()